storage/mysql/admin_storage.go

```diff
@@ -0,0 +1,394 @@
+	nowMillis := toMillisSinceEpoch(now)
+
+	newTree := *tree
+	newTree.TreeId = id
+	newTree.CreateTimeMillisSinceEpoch = nowMillis
+	newTree.UpdateTimeMillisSinceEpoch = nowMillis
+
+	insertTreeStmt, err := t.tx.Prepare(`
+		INSERT INTO Trees(
+			TreeId,
+			KeyId,
+			TreeState,
+			TreeType,
+			HashStrategy,
+			HashAlgorithm,
+			SignatureAlgorithm,
+			DuplicatePolicy,
+			DisplayName,
+			Description,
+			CreateTime,
+			UpdateTime,
+			LeafHasherType,
+			TreeHasherType,
+			AllowsDuplicateLeaves)
+		VALUES(?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)`)
+	if err != nil {
+		return nil, err
+	}
+	defer insertTreeStmt.Close()
+
+	// DuplicatePolicy doesn't map exactly to the enum, search in the map
+	// instead.
+	duplicatePolicy := ""
+	for k, v := range duplicatePolicyMap {
+		if v == newTree.DuplicatePolicy {
+			duplicatePolicy = k
+			break
+		}
+	}
+	if duplicatePolicy == "" {
+		return nil, fmt.Errorf("unexpected DuplicatePolicy value: %v", newTree.DuplicatePolicy)
+	}
+
+	_, err = insertTreeStmt.Exec(
+		newTree.TreeId,
+		1, /* KeyId */
+		newTree.TreeState.String(),
+		newTree.TreeType.String(),
+		newTree.HashStrategy.String(),
+		newTree.HashAlgorithm.String(),
+		newTree.SignatureAlgorithm.String(),
+		duplicatePolicy,
+		newTree.DisplayName,
+		newTree.Description,
+		nowDatetime,                                                            /* CreateTime */
+		nowDatetime,                                                            /* UpdateTime */
+		newTree.HashAlgorithm.String(),                                         /* LeafHasherType */
+		newTree.HashAlgorithm.String(),                                         /* TreeHasherType */
+		newTree.DuplicatePolicy == trillian.DuplicatePolicy_DUPLICATES_ALLOWED, /* AllowsDuplicateLeaves */
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	// MySQL silently truncates data when running on non-strict mode.
+	// We shouldn't be using non-strict modes, but let's guard against it
+	// anyway.
+	if _, err := t.GetTree(ctx, newTree.TreeId); err != nil {
+		// GetTree will fail for truncated enums (they get recorded as
+		// empty strings, which will not match any known value).
+		return nil, fmt.Errorf("enum truncated: %v", err)
+	}
+
+	// TODO(codingllama): There's a strong disconnect between trillian.Tree and TreeControl. Are we OK with that?
+	insertControlStmt, err := t.tx.Prepare(`
+		INSERT INTO TreeControl(
+			TreeId,
+			ReadOnlyRequests,
+			SigningEnabled,
+			SequencingEnabled,
+			SequenceIntervalSeconds,
+			SignIntervalSeconds)
+		VALUES(?, ?, ?, ?, ?, ?)`)
+	if err != nil {
+		return nil, err
+	}
+	defer insertControlStmt.Close()
+	_, err = insertControlStmt.Exec(
+		newTree.TreeId,
+		false, /* ReadOnlyRequests */
+		true,  /* SigningEnabled */
+		true,  /* SequencingEnabled */
+		defaultSequenceIntervalSeconds,
+		defaultSignIntervalSeconds,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return &newTree, nil
+}
+
+func toMillisSinceEpoch(t time.Time) int64 {
```

Comment:
Datetime stops at seconds and trillian.Tree has dates in millis_since_epoch. Nanos won't really help here.

(Plus we don't write the int64 value, we write the date-as-string one.)